server-spi-private/src/main/java/org/keycloak/utils/EmailValidationUtil.java

```diff
@@ -0,0 +1,92 @@
+package org.keycloak.utils;
+
+import java.net.IDN;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.regex.Pattern.CASE_INSENSITIVE;
+
```

Comment:
Optional: Some javadoc might be added about from where this was forked. For instance something like: `Forked from hibernate (class org.hibernate.foo.EmailValidationUtil)` . But not a blocker for this PR (just in case someone finds something else to update this)